src/command.rs

```diff
@@ -1,20 +1,142 @@
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 
 pub static HELP_STR: &str = r#"CSES CLI
 
 USAGE:
     cses-cli <command>
 
+FLAGS:
+    -h, --help          Prints this help message.
+
 COMMANDS:
     help                Prints this help message.
+    login               Log in to cses.fi
+    submit <file>       Submit a file to cses.fi.
 "#;
 
+#[derive(Debug)]
 pub enum Command {
+    None,
     Help,
+    Login,
+    Submit(String),
 }
 
 impl Command {
     pub fn from_command_line() -> Result<Command> {
-        Ok(Command::Help)
+        let pargs = pico_args::Arguments::from_env();
+
+        Command::parse_command(pargs)
+    }
+
+    fn parse_command(mut pargs: pico_args::Arguments) -> Result<Command> {
+        if pargs.contains(["-h", "--help"]) {
+            return Ok(Command::Help);
+        }
+
+        let command = pargs.subcommand()?.unwrap_or_default();
+        match command.as_str() {
+            "" => Ok(Command::None),
+            "help" => Ok(Command::Help),
+            "login" => Ok(Command::Login),
+            "submit" => {
+                let file_name = pargs.free_from_str::<String>();
```

Comment:
The type specifier is not strictly needed here, because it gets inferred from the type needed inside `Command::Submit`:

```suggestion
                let file_name = pargs.free_from_str();
```

Either way works, maybe the way it is now is slightly clearer.